components/cdc/src/initializer.rs

```diff
@@ -280,7 +272,9 @@ impl<E: KvEngine> Initializer<E> {
             }
             let cursors = old_value_cursors.as_mut();
             let resolver = resolver.as_mut();
-            let entries = self.scan_batch(&mut scanner, cursors, resolver).await?;
+            let entries = self
+                .scan_batch(&mut scanner, cursors, resolver, &mut stats)
```

Comment:
How about add start and end boundaries to the scanner?

Reply:
https://github.com/tikv/tikv/pull/17274 does such thing. But it's a huge refactor, we can't apply it to old versions. So here we just avoid unnecessary old-value loadings.